lib/rspec/core/memoized_helpers.rb

```diff
@@ -250,6 +250,9 @@ def self.hook_intention
       # available to be called from within example group blocks.
       # You can think of them as being analagous to class macros.
       module ClassMethods
+        # @private
+        ReservedKeywords = %w[initialize].freeze
```

Comment:
I think of keywords as being things like `class`, `module`, `do`, `end`, `if`, etc.  `initialize` is a reserved method.

Also, `CamelCase` suggests it's a class or module.  How about `RESERVED_METHODS = ...`?